### PR TITLE
Configure Bintray release task

### PR DESCRIPTION
**Pull request by evanw555**
Removes the old MavenDeployer setup, adds a new OSS
publishing mechanism using the Bintray Gradle plugin.
New tasks: generatePomFileForReleasePublication, bintrayUpload.

### CODE REVIEW

File: build_script/bintrayRelease.gradle [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    vcsUrl = 'https://github.com/linkedin/rest.li.git'
+    githubRepo = 'linkedin/rest.li'
+    licenses = ['Apache-2.0']
+    version {
+      name = project.version
+      released = new Date()
+      vcsTag = "v${project.version}"
+      gpg {
+        sign = true
+      }
```
Comment (mockitoguy) [inline build_script/bintrayRelease.gradle]: I suspect those may be the defaults so you don't have to declare it.

File: build_script/maven.gradle [hunk 2]
```diff
@@ -1,4 +1,5 @@
 //Configures maven plugin in a standard way
+// TODO: this isn't currently being used, delete when we're sure that we don't need it
 project.plugins.withType(MavenPlugin) {
```
Comment (mockitoguy) [inline build_script/maven.gradle]: You no longer apply "maven" plugin, so you're right, this is not needed :-)

File: gradle.properties [hunk 3]
```diff
@@ -1,6 +1,5 @@
 version=28.1.33
-sonatypeUsername=please_set_in_home_dir_if_uploading_to_maven_central
```
Comment (mockitoguy) [inline gradle.properties]: nice!

File: build_script/bintrayRelease.gradle (hunk 1 above)
Comment (evanw555) [reply build_script/bintrayRelease.gradle]: Looks like the plugin automatically picks up `project.version` as the `name` property, and the `released` property is redundant. However, the default is to _not_ sign the artifacts, so I'll need to keep that. I'll also keep the `vcsTag` property.

**Pull request merged**